Dockerfiles/data/docker-entrypoint.sh

```diff
@@ -25,14 +25,37 @@ update_uid_gid() {
 	local gid="${4}"
 
 
-	echo "[INFO] Adjusting local user with uid:${uid} and gid:${gid}"
+	echo "[INFO] Adjusting local user '${user}' with uid:${uid} and gid:${gid}"
 
 	# Remove previous user/group
 	if ! deluser "${user}"; then
 		>&2 echo "[ERR]  Failed to delete user: ${user}"
 		exit 1
 	fi
 
+	# Check if current group id already exists and fix it
```

Comment:
Good try, but I think it is an overkill. Shouldn't we just use some default IDs (e.g. `1000:1000`) or use specified uid + gid even if those are already exist in the container?
I think it is much more straightforward way instead of guessing those values.
In this case result is unpredictable.

Reply:
@snovikov thanks for the review!

You might have misunderstood this change.

Let's recap on the goal we want to achieve by running Ansible as a non-root user inside the container:


#### Current Behaviour

* You want to mount your local data into the running Docker container
* Whatever Ansible is writing to disk (e.g. via `template` module) it should do so with host uid/gid
* That's why those values can be specified as env vars via `-e MY_UID= -e MY_GID=`
* To automatically get your host uid/gid you would be using `id -u` and `id -g` respectively
    ```shell
    -e MY_UID=$(id -u) -e MY_GID=$(id -g)
    ```
* The container's startup script will then create a non-root user `ansible` with a corresponding primary group via `addgroup` and `adduser` shell command.
* The user and its group will explicitly be created with given uid/gid (so it matches host values)
* Then, whenever Ansible is run inside the container, it runs with the above created non-root user and writes data to disk with same uid/gid as your host user
* This will keep file system permissions/ownership for files/dirs of mounted volumes between container and host in sync

This is/was the current behaviour (before changes in this PR).

You can read a more detailed description about this here: https://github.com/devilbox/docker-php-fpm/blob/master/doc/syncronize-file-permissions.md


#### Issues

So what can possibly go wrong?

* Most docker images (at least the one we're using here) already ship with many default users and groups present.
* When you want to create a user or a group with a specific uid/gid that already exists, it will fail.
* This is exactly what has happened with the Ansible image during recent nightly tests


See here: https://github.com/cytopia/docker-ansible/actions/runs/3597786227/jobs/6062060091#step:12:1134

The GitHub action user on the new Ubuntu runner is using a group id of `123`. This group id already exists in the Docker image and thus, the corresponding group for the non-root user fails to create during startup.


#### Mitigation

The way I am fixing it here in this PR is as follows:

1. Check if specified uid/gid (from given env variable `MY_UID`, `MY_GID`) already exists in the container
2. If no, nothing needs to be done
3. If yes, change uid/gid of existing user/group to a value that is not yet taken, so we can free it up for our own user/group that we want to create

How is the value determined that is not yet taken?

1. The startup script checks `/etc/groups` and `/etc/passwd` inside the container to determine if uid/gid are free or already taken by another user/group
2. If that is taken, it increments the initially specified uid/gid number by one
3. Iteration is done until a uid or gid is found, that is not yet taken.
4. Once an available uid/gid is found, the overlapping user/group inside the container is moved to the new uid/gid in order to free this uid/gid for the actual non-root user/group we want to create
5. Once done, we can finally add our non-root user with a corresponding group with specified uid/gid